packages/language-server/src/plugins/typescript/DocumentSnapshot.ts

```diff
@@ -1,24 +1,216 @@
+                text = scriptInfo ? scriptInfo.content : '';
+            }
+        }
+
+        return { tsxMap, text, scriptInfo, styleInfo, parserError, nrPrependedLines };
+    }
+
+    private constructor(
+        public version: number,
+        public readonly scriptKind: ts.ScriptKind,
+        public readonly filePath: string,
+        public readonly parserError: ParserError | null,
+        public readonly scriptInfo: TagInformation | null,
+        public readonly styleInfo: TagInformation | null,
+        private readonly text: string,
+        private readonly originalText: string,
+        private readonly nrPrependedLines: number,
+        private readonly tsxMap?: RawSourceMap,
+    ) {}
+
+    getText(start: number, end: number) {
+        return this.text.substring(start, end);
+    }
+
+    getLength() {
+        return this.text.length;
+    }
+
+    getChangeRange() {
+        return undefined;
+    }
+
+    positionAt(offset: number) {
+        return positionAt(offset, this.text);
+    }
+
+    async getFragment() {
+        if (!this.fragment) {
+            const uri = pathToUrl(this.filePath);
+            const mapper = !this.scriptInfo
+                ? new IdentityMapper()
+                : !this.tsxMap
+                ? new FragmentMapper(this.originalText, this.scriptInfo)
+                : new ConsumerDocumentMapper(
+                      await new SourceMapConsumer(this.tsxMap),
+                      uri,
+                      this.nrPrependedLines,
+                  );
+            this.fragment = new SnapshotFragment(
+                mapper,
+                this.text,
+                this.scriptInfo,
+                this.styleInfo,
+                uri,
+            );
+        }
+        return this.fragment;
+    }
+}
+
+export class SnapshotFragment implements Fragment {
+    constructor(
+        private readonly mapper: DocumentMapper,
+        public readonly text: string,
+        public readonly scriptInfo: TagInformation | null,
+        public readonly styleInfo: TagInformation | null,
+        private readonly url: string,
+    ) {}
+
+    positionInParent(pos: Position): Position {
+        return this.mapper.getOriginalPosition(pos);
+    }
+
+    positionInFragment(pos: Position): Position {
+        return this.mapper.getGeneratedPosition(pos);
+    }
+
+    isInFragment(pos: Position): boolean {
+        return (
+            !this.styleInfo ||
+            !isInRange(Range.create(this.styleInfo.startPos, this.styleInfo.endPos), pos)
+        );
+    }
+
+    getURL(): string {
+        return this.url;
+    }
+
+    positionAt(offset: number) {
+        return positionAt(offset, this.text);
+    }
+
+    offsetAt(position: Position) {
+        return offsetAt(position, this.text);
     }
 }
+
+// eslint-disable-next-line
+const tsCheckRegex = /^\s*(\/\/[ \t\u00a0\u1680\u2000-\u200a\u2028\u2029\u202f\u205f\u3000\ufeff]*(@ts-(no)?check)($|\n|\r\n))/;
```

Comment:
This regex might needs a tweak, here's some examples: https://regex101.com/r/ZXpJNc/2as I think TypeScript is more liberal (any type of comment, and any whitespace after ) in accepting ts-check

Reply:
You are right. I played around a little. Seems like it accepts the comment if
- it is before the first line of code (so other lines with comments before it are ok)
- must be `@ts-(no)check`
- the comment which has `@ts-(no)check` can have any type of whitespace before it, but not other characters
- what's coming after `@ts-(no)check` is irrelevant as long there is any kind of whitespace or line break, so this would be picked up, too: `// @ts-check asdasd`